### PR TITLE
I246 contextual entities

### PR DESCRIPTION
**Pull request by balatjan**
- added support for contextual entities (csv to json direction only)

### CODE REVIEW

File: ci/unit_tests/intents_csv2json/processExample_test.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                            10,
+                            15
+                        ],
+                        "entity": "color"
+                    }
+                ]
+            }
+        result = intents_csv2json.processExample('<plant>very tall <color>green</color> tree</plant>', 'TestIntent', [])
+        assert result == expectedResult
+
```
Comment (ondrejtomas) [inline ci/unit_tests/intents_csv2json/processExample_test.py]: please try add something like:
```
assert 'omitting outer tag annotation' in self.caplog.text
```

File: ci/unit_tests/intents_csv2json/processExample_test.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+                        "location": [
+                            6,
+                            11
+                        ],
+                        "entity": "color"
+                    }
+                ]
+            }
+        result = intents_csv2json.processExample('<plant><size>small</size> <color>brown</color> pine</plant>', 'TestIntent', [])
+        assert result == expectedResult
```
Comment (ondrejtomas) [inline ci/unit_tests/intents_csv2json/processExample_test.py]: please try add something like:
```
assert 'omitting outer tag annotation' in self.caplog.text
```

File: ci/unit_tests/intents_csv2json/processExample_test.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+                        "location": [
+                            34,
+                            39
+                        ],
+                        "entity": "color"
+                    }
+                ]
+            }
+        result = intents_csv2json.processExample('<level2>hello <level1>world, how are you? this is <color>black</color></level1></level2>', 'TestIntent', [])
+        assert result == expectedResult
```
Comment (ondrejtomas) [inline ci/unit_tests/intents_csv2json/processExample_test.py]: please try add something like:
```
assert 'omitting outer tag annotation' in self.caplog.text
```

File: ci/unit_tests/intents_csv2json/processExample_test.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+    def test_intentAlreadyExisting(self):
+        ''' Test for nested entities'''
+        expectedResult = None
+        result = intents_csv2json.processExample('<hey>nothing <test>like</test> the above</hey>', 'TestIntent', [{"text": "nothing like the above"}])
+        assert result == expectedResult
```
Comment (ondrejtomas) [inline ci/unit_tests/intents_csv2json/processExample_test.py]: please try add something like:
```
assert 'Example used twice for the intent' in self.caplog.text
```

File: ci/unit_tests/intents_csv2json/processExample_test.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+    def test_intentAnnontationButEmpty(self):
+        ''' Test for nested entities'''
+        expectedResult = None
+        result = intents_csv2json.processExample('<empty></empty>', 'TestIntent', [])
+        assert result == expectedResult
```
Comment (ondrejtomas) [inline ci/unit_tests/intents_csv2json/processExample_test.py]: please try add something like:
```
assert 'Omitting empty line for intent' in self.caplog.text
```

File: ci/unit_tests/intents_csv2json/processExample_test.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+    def test_intentInvalidAnnotation(self):
+        ''' Test for nested entities'''
+        self.t_exitCode(1, ['#Doing $some <tags> @ which ~ are <not> </in> </not> correct @what?', 'TestIntent', []])
```
Comment (ondrejtomas) [inline ci/unit_tests/intents_csv2json/processExample_test.py]: please try use `t_exitCodeAndLogMessage`

Comment (balatjan) [reply ci/unit_tests/intents_csv2json/processExample_test.py]: all above fixed, thanks for comments

**Pull request merged**